File: Something_Random/services/analytics/app/routers/health.py
"""
Analytics Service — Health Route

GET /health returns service status, version, and uptime.
"""

import time
from fastapi import APIRouter

router = APIRouter()

_start_time = time.time()


@router.get("/health")
async def health_check():
    """Return current service health status."""
    return {
        "status": "ok",
        "service": "analytics",
        "version": "0.1.0",
        "timestamp": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()),
        "uptime": int(time.time() - _start_time),
    }

File: Something_Random/services/auth/src/routes/health.ts
/**
 * Auth Service — Health Route
 *
 * GET /health — Returns service health status, version, and uptime.
 */

import type { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
import type { HealthResponse } from '@nexus/types';

const startTime = Date.now();

async function healthRoute(fastify: FastifyInstance): Promise<void> {
  fastify.get(
    '/health',
    {
      schema: {
        tags: ['Health'],
        summary: 'Service health check',
        description: 'Returns the current health status of the auth service.',
        response: {
          200: {
            type: 'object',
            properties: {
              status: { type: 'string', enum: ['ok', 'degraded'] },
              service: { type: 'string' },
              version: { type: 'string' },
              timestamp: { type: 'string', format: 'date-time' },
              uptime: { type: 'number' },
            },
          },
        },
      },
    },
    async (_request: FastifyRequest, _reply: FastifyReply): Promise<HealthResponse> => {
      return {
        status: 'ok',
        service: 'auth',
        version: '0.1.0',
        timestamp: new Date().toISOString(),
        uptime: Math.floor((Date.now() - startTime) / 1000),
      };
    },
  );
}

export default healthRoute;
